### PR TITLE
Reusable base service

### PR DESCRIPTION
**Pull request by rekt-hard**
closes GH-63 

provide basic interfaces for resources and services.

### CODE REVIEW

**Comment by rekt-hard** [comment]
Currently, only person specific adaptions have taken place. If the changes are accepted, project specific adaptions will be performed.

**Comment by rekt-hard** [comment]
@ZenoLC : Should the search method on the service level only have one parameter (query parameters) and then let the implementation decide what to do? Creating a `Search` object seems like putting logic in the resource layer, when it should really be in the service layer (i.e. defining endpoints and parameters vs implementing logic).

**Comment by ZenoLC** [comment]
Yes, you are right, since we are not sending the additional information to the frontend we can just defer to the implementations discretion.

**Comment by rekt-hard** [comment]
Rebased the branch on `next`

File: src/main/java/at/ac/tuwien/damap/rest/persons/orcid/ContributorORCIDExpandedSearchMapper.java
Comment (ZenoLC) [inline src/main/java/at/ac/tuwien/damap/rest/persons/orcid/ContributorORCIDExpandedSearchMapper.java:14]: please remove this before merge, as discussed previously

**Review by ZenoLC — CHANGES_REQUESTED** [review]
Looks good and works fine. 
Only thing missing is to remove the assigning of ORCID-ID to uni-ID
And wait for frontend changes approval.

This changes the current behavior, in which a default list of projects is loaded initially. I personally would bring that back, as an additional check, but it is not within the scope of this PR, and will be made its own task.

**Review by ZenoLC — APPROVED** [review]
nice

**Pull request merged**
